src/lib.rs

```diff
@@ -1 +1,139 @@
+
+            let res = if let Some(directory) = current_directory {
+                let directory = directory.as_ref().as_os_str();
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    directory,
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            } else {
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    PWSTR::default(),
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            };
+
+            if res.as_bool() {
+                Ok(Self {
+                    command: command.to_string(),
+                    process_information: pi,
+                })
+            } else {
+                Err(ChildProcessError::CreationFailed(format!(
+                    "{:?}",
+                    GetLastError()
+                )))
+            }
+        }
+    }
+
+    pub fn wait(&self) -> ExitStatus {
+        unsafe {
+            let exit_code = WaitForSingleObject(self.process_information.hProcess, INFINITE);
+            CloseHandle(self.process_information.hProcess);
+            CloseHandle(self.process_information.hThread);
+
+            ExitStatus(exit_code)
+        }
+    }
+
+    pub fn try_wait(&self, duration: u32) -> Result<ExitStatus, ExitStatusError> {
+        unsafe {
+            match WaitForSingleObject(self.process_information.hProcess, duration) {
+                0 => Ok(ExitStatus(0)),
+                _ => Err(ExitStatusError::WaitFailed(format!("{:?}", GetLastError()))),
+            }
+        }
+    }
+
+    pub fn kill(&self) -> Result<(), ChildProcessError> {
+        unsafe {
+            if TerminateProcess(self.process_information.hProcess, 0).as_bool() {
+                Ok(())
+            } else {
+                Err(ChildProcessError::KillFailed(format!(
+                    "{:?}",
+                    GetLastError()
+                )))
+            }
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum ChildProcessError {
+    #[error("cannot create childprocess: {0}")]
+    CreationFailed(String),
+    #[error("cannot kill process: {0}")]
+    KillFailed(String),
+}
```

Comment:
Using `thiserror` helps you make clean error message for you library. As soon as you see `String` in your error type it means you most certainly failed.